osbuildconfig_webhook_trigger.yaml

```diff
@@ -0,0 +1,78 @@
+openapi: 3.0.0
+info:
+  description: OSBuild Management
+  version: 1.0.0
+  title: OSBuildManagement
+  contact:
+    name: Flotta flotta
+    url: https://github.com/project-flotta
+    email: flotta@redhat.com
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+tags:
+  - name: osbuilconfig
+    description: OSBuildConfig CRD
+paths:
+  "/namespaces/{namespace}/osbuildconfig/{name}/webhooks/{secret}/generic":
```

Comment:
what are the future plans for this API? is it expected to evolve? support multiple webhook types or will it remain as a single path?

not sure if secret should be provided in the path rather in the body.
also, what `generic` stands for?

I'd see it as
POST `/namespaces/{namespace}/osbuildconfig/{name}/triggers` with secret in requestBody.
should create a new trigger.

Reply:
take a look here: https://docs.openshift.com/container-platform/4.7/cicd/builds/triggering-builds-build-hooks.html#builds-using-generic-webhooks_triggering-builds-build-hooks
it's a reference for me, secret (value)  is part of the path
https://<openshift_api_host:port>/apis/build.openshift.io/v1/namespaces/<namespace>/buildconfigs/<name>/webhooks/<secret>/github

Reply:
No, as far as I know

Reply:
Theoretically, we can add trigger builds for osbuildconfigtemplates